### PR TITLE
TMEDIA-218 - Lokalise i18n

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Better locale file management as we work towards using Lokalise for our translations.

* All language key/values to be stored in {locale}.json files within the repository - which is also stored and sync'd in Lokalise
* Added a node script to generate each blocks intl.json file 
    * Will look at automation for this in future

{locale}.json file structure
```
{
  "test-block": {
    "translation-key": "Translation key value"
  },
  "another-blockname": {
    "translation-key": "Translation key value"
  }
}
```

## Jira Ticket
- [TMEDIA-218](https://arcpublishing.atlassian.net/browse/TMEDIA-218)

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by matthewroach** [comment]
@JackHowa I updated canary on this PR, ready for review again

File: locale/scripts/generate-intl.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    keys.forEach((item) => {
+      if (!blockOutputMap[item]) {
+        blockOutputMap[item] = {};
+      }
+
+      blockOutputMap[item][l] = fileMappings[l][key][item];
+    });
+  });
+
+  fs.writeFileSync(`./blocks/${key}/intl.json`, JSON.stringify(blockOutputMap));
```
Comment (JackHowa) [inline locale/scripts/generate-intl.js]: this is somewhat cosmetic, but it'd make validating the output json easier for me. right now, the output json isn't formatted like the previous one. I can spot-check it but hard to validate properly https://stackoverflow.com/a/3515761/7491536

Comment (JackHowa) [inline locale/scripts/generate-intl.js]:
```suggestion
  fs.writeFileSync(`./blocks/${key}/intl.json`, JSON.stringify(blockOutputMap, null, "\t"));
```

File: package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "latest-stable-parity": "lerna exec --no-bail --parallel -- npm dist-tag add \\${LERNA_PACKAGE_NAME}@$(node -e \"console.log(require('./lerna.json').version)\") stable",
     "build-storybook": "build-storybook -c .storybook -o news-theme-blocks",
     "test:coverage": "jest --coverage",
+    "generate:intl": "node locale/scripts/generate-intl.js",
```
Comment (JackHowa) [inline package.json:24]: nice. are there docs for this somewhere?

File: locale/scripts/generate-intl.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+localeFiles.forEach((file) => {
+  const lang = path.basename(file).split('.json')[0];
+  supportedLocales.push(lang);
+  fileMappings[lang] = require(resolve(file));
+});
+
+// JSON is strucutred with first key being the block name
+const blockNames = Object.keys(fileMappings[defaultLang]);
+
+blockNames.forEach((key) => {
```
Comment (JackHowa) [inline locale/scripts/generate-intl.js:22]: looks good

**Review by JackHowa — CHANGES_REQUESTED** [review]
Have a two small suggestions. but they are not necessarily big issues. since this is a spike ticket, I want to make sure we have a good base

File: package.json (hunk 2 above)
Comment (matthewroach) [reply package.json:24]: Just as part of the dropbox doc I put together.. Certainly will need to document this within confluence on how we do translations

File: blocks/global-phrases-block/intl.json [hunk 4]
```diff
@@ -1,48 +1,62 @@
...
+		"no": "Stopp",
+		"sv": "Pausa"
+	},
+	"global.gallery-autoplay-label-start": {
+		"de": "Start automatic slide show",
+		"en": "Start automatic slide show",
+		"es": "Start automatic slide show",
+		"fr": "Start automatic slide show",
+		"ja": "Start automatic slide show",
+		"ko": "Start automatic slide show",
```
Comment (JackHowa) [inline blocks/global-phrases-block/intl.json:48]: looks like this wasn't translated?

Comment (JackHowa) [reply blocks/global-phrases-block/intl.json:48]: oh, this is deliberate, right?

File: package.json (hunk 2 above)
Comment (JackHowa) [reply package.json:24]: ah gotcha

**Review by JackHowa — APPROVED** [review]
Looks good! thanks for adding formatting for readability

File: blocks/global-phrases-block/intl.json (hunk 4 above)
Comment (matthewroach) [reply blocks/global-phrases-block/intl.json:48]: Correct, this is now generating the english for missing translations

**Pull request merged**